src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,140 @@
+
+                            storage.storeTasks(tasks.getTasks());
+                        }
+                    }
+                    break;
+                case DEADLINE:
+                    if (inputSplit.length < 2) {
+                        throw new InsufficientArgumentsException(
+                                "☹ OOPS!!! The description of a deadline cannot " + "be empty!");
+                    } else {
+                        String[] arguments = inputSplit[1].split(" /by ", 2);
+                        if (arguments.length < 2) {
+                            throw new InsufficientArgumentsException("☹ OOPS!!! Missing deadline parameters!");
+                        } else {
+                            try {
+                                Task newTask = new Deadline(arguments[0], LocalDate.parse(arguments[1]));
+                                tasks.getTasks().add(newTask);
+                                Ui.printMessage(
+                                        "Got it! I've added the task:\n\t\t" + newTask.toString() + "\n\tNow you "
+                                        + "have" + " " + tasks.getTasks().size() + " tasks in " + "the list.");
+                            } catch (DateTimeParseException ex) {
+                                Ui.printMessage("☹ OOPS!!! Please enter a valid date: YYYY-MM-DD");
+                            }
+
+                            storage.storeTasks(tasks.getTasks());
+                        }
+                    }
+                    break;
+                case EVENT:
+                    if (inputSplit.length < 2) {
+                        throw new InsufficientArgumentsException(
+                                "☹ OOPS!!! The description of an event cannot" + " be empty");
+                    } else {
+                        String[] arguments = inputSplit[1].split(" /at ", 2);
+                        if (arguments.length < 2) {
+                            throw new InsufficientArgumentsException("☹ OOPS!!! Missing event parameters!");
+                        } else {
+                            try {
+                                Task newTask = new Event(arguments[0], LocalDate.parse(arguments[1]));
+                                tasks.getTasks().add(newTask);
+                                Ui.printMessage(
+                                        "Got it! I've added the task:\n\t\t" + newTask.toString() + "\n\tNow you "
+                                        + "have" + " " + tasks.getTasks().size() + " tasks in the list.");
+                            } catch (DateTimeParseException ex) {
+                                Ui.printMessage("☹ OOPS!!! Please enter a valid date: YYYY-MM-DD");
+                            }
+
+                            storage.storeTasks(tasks.getTasks());
+                        }
+                    }
+                    break;
+                case TODO:
+                    if (inputSplit.length < 2) {
```

Comment:
this can be a guard clause outside of the loop. So you can eliminate many duplicated lines